### ISSUE TITLE
Replace exceptions with another mechanism in the muparser code

### ISSUE THREAD

**Issue by krader1961** [krader1961/fish#3]
https://github.com/fish-shell/fish-shell/issues/3157 replaced our reliance on `bc` with the MuParser library. Sadly that library requires exception support be enabled. Which increases the size of fish over 230KB without even linking in the muparser library. So replace exceptions in muparser with an alternative mechanism.

**Comment by krader1961** [comment]
I've decided I don't care enough about this to bother modifying the muparser code in order to save ~200KB in binary file size.

**Comment by krader1961** [comment]
I decided to revisit this in the hope that disabling exceptions would also increase the performance as well as reduce the code size. It doesn't. Performance is identical. And while it reduces the fish binary size by 240 KB that isn't worth it. That's because the structure of the code makes it impossible to embed an error state in the `mu::Parser` class. After 12 hours of effort I gave up. I spent another three hours implementing a global error state just so I could measure the impact of disabling exceptions. And at that point errors were not handled correctly because I had not updated all the `try...catch` blocks. I estimate another two to three days of work to ensure errors are correctly handled.
